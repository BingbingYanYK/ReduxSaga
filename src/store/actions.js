export const FIND_PAYMENT = {
  REQUEST: 'payment/FIND/REQUEST',
  SUCCESS: 'payment/FIND/SUCCESS',
  FAILURE: 'payment/FIND/FAILURE'
};

export const SET_DESTINATION = 'order/SET_DESTINATION';
export const SET_PAYMENT = 'order/SET_PAYMENT';
export const SUBMIT = 'order/SUBMIT';

export function setPayment(paymentId) {
  return {
    type: SET_PAYMENT,
    paymentId
  };
}

export function startTrip() {
  return {
    type: SUBMIT
  };
}
