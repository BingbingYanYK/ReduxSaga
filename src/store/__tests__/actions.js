import * as actions from '../actions';

describe('store/actions', () => {
  describe('#setPayment', () => {
    it('creates the correct action', () => {
      const action = actions.setPayment(0);
      expect(action).toMatchSnapshot();
    });
  });
  describe('#startTrip', () => {
    it('creates the correct action', () => {
      const action = actions.startTrip();
      expect(action).toMatchSnapshot();
    });
  });
});
