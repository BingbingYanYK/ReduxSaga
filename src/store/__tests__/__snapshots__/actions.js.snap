// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/actions #setPayment creates the correct action 1`] = `
Object {
  "paymentId": 0,
  "type": "order/SET_PAYMENT",
}
`;

exports[`store/actions #startTrip creates the correct action 1`] = `
Object {
  "type": "order/SUBMIT",
}
`;
